gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -219,6 +225,101 @@ impl GatewayLightning for ClnRpcService {
             })
     }
 
+    async fn get_route_hints(
+        &self,
+        _request: tonic::Request<EmptyRequest>,
+    ) -> Result<tonic::Response<GetRouteHintsResponse>, Status> {
+        let our_pub_key = self
+            .pubkey()
+            .await
+            .map_err(|err| tonic::Status::internal(err.to_string()))?;
+
+        let mut client = self
+            .rpc_client()
+            .await
+            .map_err(|err| tonic::Status::internal(err.to_string()))?;
+
+        let peers_response = client
+            .call(cln_rpc::Request::ListPeers(model::ListpeersRequest {
+                id: None,
+                level: None,
+            }))
+            .await
+            .map_err(|err| tonic::Status::internal(err.to_string()))?;
+
+        let peers = match peers_response {
+            cln_rpc::Response::ListPeers(peers) => Ok(peers.peers),
+            _ => Err(ClnExtensionError::RpcWrongResponse),
+        }
+        .map_err(|err| tonic::Status::internal(err.to_string()))?;
+
+        let active_peer_channels = peers
+            .into_iter()
+            .flat_map(|peer| peer.channels.into_iter().map(move |chan| (peer.id, chan)))
+            .filter_map(|(peer_id, chan)| {
+                // TODO: upstream eq derive
+                if !matches!(
+                    chan.state,
+                    model::ListpeersPeersChannelsState::CHANNELD_NORMAL
+                ) {
+                    return None;
+                }
```

Comment:
that seems a bit verbose given that we don't have multiple possible matches but only one. 
I think something like this should work? 

```rust
if let ListpeersPeersChannelState::CHANNELD_NORMAL = chan.state {
...
}
```